c-dependencies/js-compute-runtime/xqd-world/xqd_world_adapter.cc

```diff
@@ -0,0 +1,966 @@
+#include "xqd_world_adapter.h"
+
+JSContext *context = nullptr;
+
+void set_js_context(JSContext *cx) { context = cx; }
+
+#ifndef COMPONENT
+
+// Ensure that all the things we want to use the hostcall buffer for actually
+// fit into the buffer.
+#define HOSTCALL_BUFFER_LEN HEADER_MAX_LEN
+static_assert(DICTIONARY_ENTRY_MAX_LEN < HOSTCALL_BUFFER_LEN);
+static_assert(METHOD_MAX_LEN < HOSTCALL_BUFFER_LEN);
+static_assert(URI_MAX_LEN < HOSTCALL_BUFFER_LEN);
+
+extern "C" {
+
+__attribute__((weak, export_name("cabi_realloc"))) void *
+cabi_realloc(void *ptr, size_t orig_size, size_t align, size_t new_size) {
+  return JS_realloc(context, ptr, orig_size, new_size);
+}
+}
+
+#define LIST_ALLOC_SIZE 50
+
+static fastly_error_t convert_result(int res) {
+  switch (res) {
+  case 0:
+    return FASTLY_RESULT_ERROR_OK;
+  case 1:
+    return FASTLY_ERROR_GENERIC_ERROR;
+  case 2:
+    return FASTLY_ERROR_INVALID_ARGUMENT;
+  case 3:
+    return FASTLY_ERROR_BAD_HANDLE;
+  case 4:
+    return FASTLY_ERROR_BUFFER_LEN;
+  case 5:
+    return FASTLY_ERROR_UNSUPPORTED;
+  case 6:
+    return FASTLY_ERROR_BAD_ALIGN;
+  case 7:
+    return FASTLY_ERROR_HTTP_INVALID;
+  case 8:
+    return FASTLY_ERROR_HTTP_USER;
+  case 9:
+    return FASTLY_ERROR_HTTP_INCOMPLETE;
+  case 10:
+    return FASTLY_ERROR_OPTIONAL_NONE;
+  case 11:
+    return FASTLY_ERROR_HTTP_HEAD_TOO_LARGE;
+  case 12:
+    return FASTLY_ERROR_HTTP_INVALID_STATUS;
```

Comment:
This is missing the newer error case for exceeding limits (https://github.com/fastly/Viceroy/commit/8ee547b53cadb8366082bd7ca931be0a3a770411):
```suggestion
  case 12:
    return FASTLY_ERROR_HTTP_INVALID_STATUS;
  case 13:
    return FASTLY_ERROR_LIMIT_EXCEEDED;
```